packages/internal/src/kv-store.js

```diff
@@ -3,37 +3,26 @@ import { Fail } from '@endo/errors';
 
 /**
  * @typedef {{
- *   has: (key: string) => boolean,
- *   get: (key: string) => string | undefined,
- *   getNextKey: (previousKey: string) => string | undefined,
- *   set: (key: string, value: string, bypassHash?: boolean ) => void,
- *   delete: (key: string) => void,
+ *   has: (key: string) => boolean;
+ *   get: (key: string) => string | undefined;
+ *   getNextKey: (previousKey: string) => string | undefined;
+ *   set: (key: string, value: string, bypassHash?: boolean) => void;
+ *   delete: (key: string) => void;
  * }} KVStore
  */
 
 /**
- * @param {string} key
+ * @param {object} db The SQLite database connection.
+ * @param {() => void} ensureTxn Called before mutating methods to establish a
+ *   DB transaction
+ * @param {(...args: string[]) => void} trace Called after set/delete to record
+ *   a debug log
+ * @returns {KVStore}
  */
-export function getKeyType(key) {
-  typeof key === 'string' || Fail`key must be a string`;
-  if (key.startsWith('local.')) {
-    return 'local';
-  } else if (key.startsWith('host.')) {
-    return 'host';
-  }
-  return 'consensus';
-}
 
-/**
- * @param {object} db  The SQLite database connection.
- * @param {() => void} ensureTxn  Called before mutating methods to establish a DB transaction
- * @param {(...args: string[]) => void} trace  Called after set/delete to record a debug log
- * @returns { KVStore }
- */
-
-export function makeKVStore(db, ensureTxn, trace) {
+export function makeKVStore(db, ensureTxn, trace, tableName = 'kvStore') {
```

Comment:
I don't think the new resolver use case justifies this generalization. Let's just stick with the non-configurable "kvStore" table name until we actually have a need for something else.
```suggestion
export function makeKVStore(db, ensureTxn, trace) {
```